src/QuadForm/Torsion.jl

```diff
@@ -650,3 +649,312 @@ function normal_form(T::TorQuadMod; partial=false)
   end
   return sub(T, normal_gens)
 end
+
+@doc Markdown.doc"""
+_brown_indecomposable(q::MatElem, p::fmpz) ->  fmpz
+Return the Brown invariant of the indecomposable form ``q``.
+
+The values are taken from Table 2.1 in [Shim2016]_.
+INPUT:
+- ``q`` - an indecomposable quadratic form represented by a
+  rational `1 \times 1` or `2 \times 2` matrix
+- ``p`` - a prime number
+EXAMPLES::
+
+  julia> q = matrix(QQ, 1, 1, [1//3])
+  julia> _brown_indecomposable(q,fmpz(3))
+  6
+  julia> q = matrix(QQ, 1, 1, [2//3])
+  julia> _brown_indecomposable(q,fmpz(3))
+  2
+"""
+function _brown_indecomposable(q::MatElem, p::fmpz)
+  v = valuation(denominator(q), p)
+  if p == 2
+    # brown(U) = 0
+    if ncols(q) == 2
+      if valuation(q[1,1],2) > v + 1 && valuation(q[2, 2],2) > v + 1
+        # type U
+        return mod(0, 8)
+      else
+        # type V
+        return mod(4*v, 8)
+      end
+    end
+    u = numerator(q[1, 1])
+    return mod(divexact(u + v*(u^2 - 1), 2), 8)
```

Comment:
The tests are failing because this `divexact` is failing. Apparently `2` is not dividing `u + v*(u^2 - 1)`. Is this possible?

Reply:
I believe it is possible. However, the tests are not failing when I try them locally on my system. I'm not sure why it is failing. I'll try updating the branch and do another pull request